[quant] Add quantizer class skeleton

asl3 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79845
* #79940
* #79530
* #79499
* __->__ #79936

### Summary:
This PR adds a quantizer class skeleton to store APoT quantize and dequantize methods, as well as APoT quantization parameters.

### Test Plan:
Run unit tests with: `python pytorch/test/quantization/core/experimental/test_quantizer.py`

asl3 committed:
[quant] Add quantizer class skeleton

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantizer class skeleton"




[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantizer class skeleton"


Summary:
This PR adds a quantizer class skeleton to store APoT quantize and dequantize methods, as well as APoT quantization parameters.

Test Plan:
Run unit tests with: python pytorch/test/quantization/core/experimental/test_quantizer.py

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantizer class skeleton"


Summary:
This PR adds a quantizer class skeleton to store APoT quantize and dequantize methods, as well as APoT quantization parameters.

Test Plan:
Run unit tests with: python pytorch/test/quantization/core/experimental/test_quantizer.py

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantizer class skeleton"


Summary:
This PR adds a quantizer class skeleton to store APoT quantize and dequantize methods, as well as APoT quantization parameters.

Test Plan:
Run unit tests with: python pytorch/test/quantization/core/experimental/test_quantizer.py

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantizer class skeleton"


Summary:
This PR adds a quantizer class skeleton to store APoT quantize and dequantize methods, as well as APoT quantization parameters.

Test Plan:
Run unit tests with: python pytorch/test/quantization/core/experimental/test_quantizer.py

[ghstack-poisoned]

dzdang commented:
PR name and summar yshould be changed

jerryzh168 commented on torch/ao/quantization/experimental/quantizer.py:
```diff
@@ -0,0 +1,16 @@
+# class to store APoT quantizer
+# implements quantize and dequantize
+# and stores all quantization parameters
+class APoTQuantizer(torch.Tensor):
```
nit: quantizer doesn't need to inherit from Tensor, it's a separate object

jerryzh168 reviewed (approved):
looks good, had one comment inline

asl3 committed:
Update on "[quant] Add quantizer class skeleton"


### Summary:
This PR adds a quantizer class skeleton to store APoT quantize and dequantize methods, as well as APoT quantization parameters.

### Test Plan:
Run unit tests with: `python pytorch/test/quantization/core/experimental/test_quantizer.py`

[ghstack-poisoned]

asl3 commented:
@pytorchbot merge -g

asl3 committed:
Update on "[quant] Add quantizer class skeleton"


### Summary:
This PR adds a quantizer class skeleton to store APoT quantize and dequantize methods, as well as APoT quantization parameters.

### Test Plan:
Run unit tests with: `python pytorch/test/quantization/core/experimental/test_quantizer.py`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantizer class skeleton"


### Summary:
This PR adds a quantizer class skeleton to store APoT quantize and dequantize methods, as well as APoT quantization parameters.

### Test Plan:
Run unit tests with: `python pytorch/test/quantization/core/experimental/test_quantizer.py`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantizer class skeleton"


### Summary:
This PR adds a quantizer class skeleton to store APoT quantize and dequantize methods, as well as APoT quantization parameters.

### Test Plan:
Run unit tests with: `python pytorch/test/quantization/core/experimental/test_quantizer.py`

[ghstack-poisoned]

asl3 commented:
@pytorchbot merge -g

asl3 committed:
Update on "[quant] Add quantizer class skeleton"


### Summary:
This PR adds a quantizer class skeleton to store APoT quantize and dequantize methods, as well as APoT quantization parameters.

### Test Plan:
Run unit tests with: `python pytorch/test/quantization/core/experimental/test_quantizer.py`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantizer class skeleton"


### Summary:
This PR adds a quantizer class skeleton to store APoT quantize and dequantize methods, as well as APoT quantization parameters.

### Test Plan:
Run unit tests with: `python pytorch/test/quantization/core/experimental/test_quantizer.py`

[ghstack-poisoned]

asl3 commented:
@pytorchbot merge -g

asl3 committed:
Update on "[quant] Add quantizer class skeleton"


### Summary:
This PR adds a quantizer class skeleton to store APoT quantize and dequantize methods, as well as APoT quantization parameters.

### Test Plan:
Run unit tests with: `python pytorch/test/quantization/core/experimental/test_quantizer.py`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantizer class skeleton"


### Summary:
This PR adds a quantizer class skeleton to store APoT quantize and dequantize methods, as well as APoT quantization parameters.

### Test Plan:
Run unit tests with: `python pytorch/test/quantization/core/experimental/test_quantizer.py`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantizer class skeleton"


### Summary:
This PR adds a quantizer class skeleton to store APoT quantize and dequantize methods, as well as APoT quantization parameters.

### Test Plan:
Run unit tests with: `python pytorch/test/quantization/core/experimental/test_quantizer.py`

[ghstack-poisoned]

asl3 committed:
Update on "[quant] Add quantizer class skeleton"


### Summary:
This PR adds a quantizer class skeleton to store APoT quantize and dequantize methods, as well as APoT quantization parameters.

### Test Plan:
Run unit tests with: `python pytorch/test/quantization/core/experimental/test_quantizer.py`

[ghstack-poisoned]

asl3 commented:
git merge -g

asl3 commented:
@pytorchbot merge -g
